src/bundler.ts

```diff
@@ -24,6 +25,54 @@ interface BundleOptions {
   onBeforeDownload?: LifecycleHook
 }
 
+interface BundleOpsArgs {
+  buildID: string
+  debug?: boolean
+  deno: DenoBridge
+  basePath: string
+  distDirectory: string
+  functions: EdgeFunction[]
+  importMap: ImportMap
+  featureFlags: Record<string, string>
+}
+
+const createBundleOps = ({
+  basePath,
+  buildID,
+  debug,
+  deno,
+  distDirectory,
+  functions,
+  importMap,
+  featureFlags,
+}: BundleOpsArgs) => {
+  const bundleOps = [
+    bundleJS({
+      buildID,
+      debug,
+      deno,
+      distDirectory,
+      functions,
+      importMap,
+    }),
+  ]
+
+  if (featureFlags.edge_functions_produce_eszip) {
+    bundleOps.push(
+      bundleESZIP({
+        basePath,
+        buildID,
+        debug,
+        deno,
+        distDirectory,
+        functions,
+      }),
+    )
+  }
+
+  return bundleOps
+}
+
```

Comment:
I shifted this into its own function because the linter was complaining (rightly so) about the function getting too long. I'm not sure about the name `BundleOpsArgs` though, sounds kinda vague?

Reply:
I like it! Great shout, I'll have a play around with it now